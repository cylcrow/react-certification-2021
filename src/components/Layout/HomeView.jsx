import React from 'react';
import styled from 'styled-components';

const StyledHomeView = styled.div`
  padding-top: 64px;
  height: calc(100% - 64px);
`;

const HomeView = () => {
  return (
    <StyledHomeView data-testid="main-content">
      I am your main content, brother.
    </StyledHomeView>
  );
};

export default HomeView;
