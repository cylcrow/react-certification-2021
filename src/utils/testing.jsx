import React from "react";
import { render } from '@testing-library/react';
import { ThemeProvider } from "styled-components";
import { lightTheme } from '../providers/theme/themes';

export const renderWithTheme = (Component, theme = lightTheme) => {
    return render(
        <ThemeProvider theme={{ theme, switchTheme: jest.fn() }}>
        { Component }
        </ThemeProvider>
    );
}