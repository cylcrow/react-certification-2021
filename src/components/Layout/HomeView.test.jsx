import React from 'react';
import '../../utils/testing';
import 'jest-styled-components';
import { render, unmountComponentAtNode } from 'react-dom';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import HomeView from './HomeView';

let container;
const build = () => {
  render(<HomeView />, container);
};

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  return container;
});

describe('HomeView', () => {
  it('renders', () => {
    const wrapper = build(<HomeView />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('HomeView styles', () => {
  it('applies default styling', () => {
    const tree = mount(<HomeView />);
    expect(tree).toHaveStyleRule('padding-top', '64px');
    expect(tree).toHaveStyleRule('height', 'calc(100% - 64px)');
  });
});
