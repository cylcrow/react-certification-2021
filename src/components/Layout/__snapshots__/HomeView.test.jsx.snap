// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView renders 1`] = `undefined`;
